ShopLens/ShopLensForms/Controllers/MainController.cs

```diff
@@ -0,0 +1,165 @@
+﻿using ImageRecognition.Classificators;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+using VoicedText.TextVoicers;
+using VoiceRecognitionWithTextVoicer.VoiceRecognizers;
+
+namespace ShopLensForms.Controllers
+{
+    public class MainController
+    {
+        /// <inheritdoc cref="ITextVoicer"/>
+        private ITextVoicer _textVoicer;
+
+        /// <inheritdoc cref="IVoiceRecognizer"/>
+        private IVoiceRecognizer _voiceRecognizer;
+
+        /// <inheritdoc cref="IImageClassificator"/>
+        private IImageClassificator _imageClassifying;
+
+        /// <summary>
+        /// A windows form that the controller communicates with.
+        /// </summary>
+        public IntroForm _introForm;
+
+        /// <inheritdoc cref="_introForm"/>
+        public ShopLens _shopLens;
+
+        private const string helloCmd = "Hello";
+        private const string whatIsThisCmd = "What is this";
+        private const string startCmd = "Start";
+        private const string exitCmd = "Exit";
+
+        public MainController()
+        {
+            _textVoicer = new TextVoicerSpeechSynthesizer();
+            _voiceRecognizer = new VoiceRecognizerSpeechRecEngine();
+            _imageClassifying = new TensorFlowClassificator();
+
+            _introForm = new IntroForm(this);
+            _shopLens = new ShopLens(this);
+        }
+
+        [STAThread]
+        public void StartApp()
+        {
+            StartVoiceRecognizer();
+            Application.Run(_introForm);
+        }
+
+        public void StartVoiceRecognizer()
+        {
+            _voiceRecognizer.AddCommand(helloCmd, CommandRecognized_Hello);
+            _voiceRecognizer.AddCommand(whatIsThisCmd, CommandRecognized_WhatIsThis);
+            _voiceRecognizer.AddCommand(startCmd, CommandRecognized_Start);
+            _voiceRecognizer.AddCommand(exitCmd, CommandRecognized_Exit);
+            _voiceRecognizer.StartVoiceRecognition();
+        }
+
+        /// <summary> Calls method when someone says a certain command. </summary>
+        /// <remarks>
+        /// This if statement makes sure the method that must be executed when the specified command is recognized
+        /// is called within the GUI thread. For information see https://stackoverflow.com/a/10170699.
+        /// </remarks>
+        private void CommandRecognized_Hello(object sender, EventArgs e)
+        {
+            if (_introForm.InvokeRequired)
```

Comment:
Not very DRY 🌦 .

🏜 = perhaps create a method like `InvokeOnMainThread` instead of writing "if ... begin invoke... else ..." in every event.

Reply:
I'll make it as DRY as the Atacama Desert.